src/components/app-configure/access/CreateDelegate.vue

```diff
@@ -0,0 +1,161 @@
+<script lang="ts" setup>
+import { ref } from 'vue'
+
+import VButton from '@/components/lib/VButton/VButton.vue'
+import VCard from '@/components/lib/VCard/VCard.vue'
+import VDropdown from '@/components/lib/VDropdown/VDropdown.vue'
+import VOverlay from '@/components/lib/VOverlay/VOverlay.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import { useToast } from '@/components/lib/VToast'
+import { createDelegate } from '@/services/gateway.service'
+import { useAppId } from '@/use/getAppId'
+
+const emit = defineEmits(['close', 'generateKey', 'created'])
+
+const props = defineProps({
+  delegateKeys: {
+    type: Array,
+    required: true,
+  },
+})
```

Comment:
Use TypeScript to define the prop types. Vue's object syntax for defining props is quite limited — `Array`, for example, could be an array of anything.

Reply:
https://github.com/arcana-network/developer-dashboard/pull/156/commits/764ff0c48c67ed42721b0494b417f70842081ebb